scriptmodules/supplementary/wifi.sh

```diff
@@ -142,13 +147,16 @@ function gui_connect_wifi() {
         id=$(iwgetid -r)
         ((i++))
     done
-    [[ -z "$id" ]] && printMsgs "dialog" "Unable to connect to network $essid"
+    if [[ -z "$id" ]]; then
+        printMsgs "dialog" "Unable to connect to network $essid"
+        ifdown wlan0 &>/dev/null || ip link set wlan0 down &>/dev/null
+    fi
 }
 
 function gui_wifi() {
     local default
     while true; do
-        local ip_int=$(ip route get 8.8.8.8 2>/dev/null | head -1 | cut -d' ' -f8)
+        local ip_int=$(ip route ls dev wlan0 2>/dev/null | awk 'END {print $7}')
```

Comment:
This should show the current IP, not the current wlan0 interface ip. eg an ethernet cable could be connected. I assume the above change would not show the LAN ip if wifi was off ?

Reply:
if we are going to switch to showing just the wlan0 ip address, the wording should change. Or maybe we can show both. Not sure I want this change for now though.

Reply:
>Or maybe we can show both

I like the idea of showing both. granted showing ethernet stuff in the wifi module is a bit misleading, or at the very least have it show them with the showip module in the retropie menu. perhaps the module should have been called networking instead of wifi ;)

Reply:
I'm happy to either clarify the text or revert.

Maybe we could have:
```
local ip_int $(ip route ls dev wlan0 2>/dev/null | awk 'END {print $7}')
local ip_route $(ip route get 8.8.8.8 2>/dev/null | awk '{print $NF; exit}')"
```
Then print:
```
Current wireless IP: $ip_int (default route: $ip_route)
```

Better or just leave it to the previous behaviour?

Reply:
I'll merge as is for now.

Reply:
Happy for your change above, although ip_wlan would be better var name no (int = internal ? the ethernet ip address is internal also).